src/scripts/main.js

```diff
@@ -1,3 +1,337 @@
+  form.setAttribute('id', 'dynamicForm');
+  form.setAttribute('action', '/submit');
+  form.setAttribute('method', 'post');
+
+  // Add Name field
+  const nameLabel = createLabel('Name: ', 'name');
+  const nameInput = createInput('name', 'text');
+
+  nameLabel.append(' ', nameInput);
+  form.append(nameLabel);
+
+  // Add Position field
+  const positionLabel = createLabel('Position: ', 'position');
+  const positionInput = createInput('position', 'text');
+
+  positionLabel.append(' ', positionInput);
+  form.append(positionLabel);
+
+  // Add Office (City) field
+  const selectLabel = createLabel('Office: ', '');
+  const officeSelect = createSelect(CITIES);
+
+  officeSelect.setAttribute('name', 'city');
+  officeSelect.setAttribute('data-qa', 'office');
+  selectLabel.append(officeSelect);
+  form.append(selectLabel);
+
+  // Add Age field
+  const ageLabel = createLabel('Age: ', 'age');
+  const ageInput = createInput('age', 'number');
+
+  ageLabel.append(ageInput);
+  form.append(ageLabel);
+
+  // Add Salary field
+  const salaryLabel = createLabel('Salary: ', 'salary');
+  const salaryInput = createInput('salary', 'number');
+
+  salaryLabel.append(salaryInput);
+  form.append(salaryLabel);
+
+  // Add Save Button
+  const saveButton = createSubButton('dynamicForm', 'Save to table');
+
+  saveButton.addEventListener('click', handleFormSubmit);
+  form.append(saveButton);
+
+  // Append form to the document body
+  document.body.append(form);
+}
+
+// Handles the form submission, validates data, and adds a new row to the table
+function handleFormSubmit(e) {
+  e.preventDefault(); // Prevent form's default behavior (refresh)
+
+  const form = document.getElementById('dynamicForm');
+  const nameValue = form.elements['name'].value;
+  const positionValue = form.elements['position'].value;
+  const selectedOffice = form.elements['city'].value;
+  const ageValue = Number(form.elements['age'].value);
+  const salaryValue = Number(form.elements['salary'].value);
+
+  // Form validation checks
+  if (!form.checkValidity()) {
+    showNotification('error', 'Please fill in all required fields.');
+
+    return;
+  }
+
+  if (nameValue.length < MIN_NAME_LENGTH) {
+    showNotification('error', 'Name must be at least 4 characters long.');
+
+    return;
+  }
+
+  if (ageValue < MIN_AGE || ageValue > MAX_AGE) {
+    showNotification('error', 'Age must be between 18 and 90.');
+
+    return;
+  }
+
+  // Create new employee row and add to table
+  const newEmployee = createNewEmployee(
+    nameValue,
+    positionValue,
+    selectedOffice,
+    ageValue,
+    salaryValue,
+  );
+
+  // Notify user of successful form submission
+  showNotification('success', 'Form submitted successfully!');
+  tableBody.append(newEmployee);
+
+  // Reset the form fields
+  form.reset();
+}
+
+// Creates a new employee row with the provided data
+function createNewEmployee(employeName, position, office, age, salary) {
```

Comment:
There is a typo in the parameter name `employeName`. It should be `employeeName` to maintain consistency and clarity.